pulser-core/pulser/errors/__init__.py

```diff
@@ -0,0 +1,301 @@
+"""Errors raised by Pulser.
```

Comment:
Why use a module with just an `__init__.py` inside of a simple `errors.py` file?

Reply:
Also, I imagine we'll want to include this module in the API reference, right?

Reply:
> Why use a module with just an `__init__.py` inside of a simple `errors.py` file?

In early prototypes, the module was spread across several submodules, so having an `__init__.py` made sense, but there's no good reason anymore. Happy to move this to an `errors.py` if you feel it's better.


> Also, I imagine we'll want to include this module in the API reference, right?

Absolutely. I'm not familiar with how the Pulser API is generated, how do I do that?

Reply:
> In early prototypes, the module was spread across several submodules, so having an `__init__.py` made sense, but there's no good reason anymore. Happy to move this to an `errors.py` if you feel it's better.

Would it be okay to call it `exceptions.py` instead? I find the name less ambiguous

> Absolutely. I'm not familiar with how the Pulser API is generated, how do I do that?

In principle, adding the module [here](https://github.com/pasqal-io/Pulser/blob/f1bd30b7d8a5aadd1e06e3c4d12513584115c316/docs/source/apidoc/core.rst?plain=1#L61) should do it.
Also, I guess we'll want to export the module in the `pulser-core` too